### PR TITLE
[Merged by Bors] - feat: Minimal axioms for rings and fields

### PR DESCRIPTION
**Pull request by ChrisHughes24**
---
Follow up to #6173 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Field/MinimalAxioms.lean [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    (mul_inv_cancel : ∀ a : K, a ≠ 0 → a * a⁻¹ = 1)
+    (inv_zero : (0 : K)⁻¹ = 0)
+    (left_distrib : ∀ a b c : K, a * (b + c) = a * b + a * c)
+    (exists_pair_ne : ∃ x y : K, x ≠ y) : Field K := by
+  letI := CommRing.ofMinimalAxioms add_assoc zero_add
+    add_left_neg mul_assoc mul_comm one_mul left_distrib
+  exact
+  { exists_pair_ne := exists_pair_ne
+    mul_inv_cancel := mul_inv_cancel
+    inv_zero := inv_zero }
```
Comment (eric-wieser) [inline Mathlib/Algebra/Field/MinimalAxioms.lean:43]:
```suggestion
    (exists_pair_ne : ∃ x y : K, x ≠ y) : Field K :=
  letI := CommRing.ofMinimalAxioms add_assoc zero_add
    add_left_neg mul_assoc mul_comm one_mul left_distrib
  { exists_pair_ne := exists_pair_ne
    mul_inv_cancel := mul_inv_cancel
    inv_zero := inv_zero }
```

**Comment by ericrbg** [comment]
should we make it more clear that any structure created with these would undoubtedly lead to diamonds?

**Comment by ChrisHughes24** [comment]
> should we make it more clear that any structure created with these would undoubtedly lead to diamonds?

I think they're no more dangerous than using the existing constructors and allowing the `nsmul` etc. to be given automatically.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**